test/rtp_decoder.c

```diff
@@ -671,32 +702,51 @@ void rtp_decoder_handle_pkt(u_char *arg,
     }
     const void *rtp_packet = bytes + dcdr->rtp_offset;
 
-    memcpy((void *)&dcdr->message, rtp_packet, hdr->caplen - dcdr->rtp_offset);
+    memcpy((void *)&message, rtp_packet, hdr->caplen - dcdr->rtp_offset);
     pktsize = hdr->caplen - dcdr->rtp_offset;
     octets_recvd = pktsize;
 
     if (octets_recvd == -1) {
         return;
     }
 
-    /* verify rtp header */
-    if (dcdr->message.header.version != 2) {
-        return;
+    if (dcdr->mode == mode_rtp) {
+        rtp = 1;
+    } else if (dcdr->mode == mode_rtcp) {
+        rtp = 0;
+    } else {
+        rtp = 1;
+        if (octets_recvd >= 2) {
+            /* rfc5761 */
+            u_char pt = *(bytes + dcdr->rtp_offset + 1) & 0x7f;
```

Comment:
Maybe use `payload_type` to be a bit more explicit? Makes RFC perusing a bit easier. :-)